titiler/stacapi/factory.py

```diff
@@ -195,35 +194,32 @@ def tile(
             scale = scale or 1
 
             tms = self.supported_tms.get(tileMatrixSetId)
-            with rasterio.Env(**env):
```

Comment:
I removed this context manager because I saw one was being used in AssetReader#tile#_reader `with self.ctx` but should we have it here as well since as you mentioned there are 3 levels of reader (STAC, item and asset)?

Reply:
it's intentional to have an env for each reader, so we can set variable like `GDAL_INGESTED_BYTES_AT_OPEN` using specific number